vhost-device-sound/src/stream.rs

```diff
@@ -239,85 +244,92 @@ impl Default for PcmParams {
     }
 }
 
-pub struct Buffer {
-    data_descriptor: virtio_queue::Descriptor,
+pub struct Request {
+    //data_descriptor: virtio_queue::Descriptor,
     pub pos: usize,
+    len: usize,
     pub message: Arc<IOMessage>,
     direction: Direction,
 }
 
-impl std::fmt::Debug for Buffer {
+impl std::fmt::Debug for Request {
     fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
-        fmt.debug_struct(stringify!(Buffer))
+        fmt.debug_struct(stringify!(Request))
+            .field("len", &self.len)
             .field("pos", &self.pos)
```

Comment:
I would try to keep these ordered the same way as they are in the definition. That is of course violated already, but maybe we should not scramble it further? :)